ddl/partition.go

```diff
@@ -761,6 +763,15 @@ func checkPartitioningKeysConstraints(sctx sessionctx.Context, s *ast.CreateTabl
 	return nil
 }
 
+func hasPrefixIndexColumn(idxCols []*model.IndexColumn) bool {
+	for _, col := range idxCols {
+		if col.Length > 0 {
```

Comment:
Should check the table name too?

Below sql will execute sucessfull in Mysql but faild in This PR:
```sql
create table part12 (a varchar(20), b varchar(10), unique index (a(5),b)) partition by range columns (b) (
			partition p0 values less than ('aaaaa'),
			partition p1 values less than ('bbbbb'),
			partition p2 values less than ('ccccc'))
```

Reply:
Fixed